pupil-api/src/services/pupil-authentication.service.ts

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+import * as azureStorage from 'azure-storage'
+import * as bluebird from 'bluebird'
+import { clone } from 'ramda'
+import * as winston from 'winston'
+
+let azureTableService: any
+const authTable = 'preparedCheck'
+
+export const pupilAuthenticationService = {
+  authenticate: async function authenticate (pupilPin: string, schoolPin: string, tableService?: any): Promise<object> {
+    // set tableService to azureTableService if not provided, but only instantiate it once.
+    if (!tableService) {
+      if (!azureTableService) {
+        azureTableService = azureStorage.createTableService()
+        bluebird.promisifyAll(azureTableService, {
+          promisifier: (originalFunction) => function (...args) {
+            return new Promise((resolve, reject) => {
+              try {
+                originalFunction.call(this, ...args, (error, result, response) => {
+                  if (error) {
+                    return reject(error)
+                  }
+                  resolve({ result, response })
+                })
+              } catch (error) {
+                reject(error)
+              }
+            })
+          }
+        })
+      }
+      tableService = azureTableService
```

Comment:
This might look nicer/readable in a separate `initTableService()` function or similar?